### PR TITLE
[IMP] sale_pdf_quote_builder: improve feature

### PR DESCRIPTION
**Pull request by Demesmaeker**
Following https://github.com/odoo/odoo/commit/116879e17e81657f48a7d11780d8d30715ecc68f a few improvements were needed to make it more
complete.

task-3484125

### CODE REVIEW

File: odoo/tools/pdf.py [hunk 1]
```diff
@@ -77,35 +77,51 @@ def __init__(self):
...
-                        writer.updatePageFormFieldValues(page, form_fields)
+            writer.addPage(reader.getPage(page))
 
     with io.BytesIO() as _buffer:
         writer.write(_buffer)
         return _buffer.getvalue()
 
+def fill_form_fields_pdf(document, form_fields=None):
+    ''' Merge a collection of PDF documents in one.
+    Note that the attachments are not merged.
```
Comment (Feyensv) [inline odoo/tools/pdf.py:98]: Docstring c/p but this part has not been adapted ;)

File: addons/sale_pdf_quote_builder/wizards/res_config_settings.py [hunk 2]
```diff
@@ -7,4 +7,6 @@ class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
     sale_header = fields.Binary(related='company_id.sale_header', readonly=False)
+    sale_header_name = fields.Char(related='company_id.sale_header_name')
     sale_footer = fields.Binary(related='company_id.sale_footer', readonly=False)
+    sale_footer_name = fields.Char(related='company_id.sale_footer_name')
```
Comment (Feyensv) [inline addons/sale_pdf_quote_builder/wizards/res_config_settings.py:12]: Should be hopefully removed later when we have a cleaner way to display the filename.

File: addons/sale_pdf_quote_builder/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+    # Part of Odoo. See LICENSE file for full copyright and licensing details.
```
Comment (Feyensv) [inline addons/sale_pdf_quote_builder/models/sale_order.py:1]:
```suggestion
# Part of Odoo. See LICENSE file for full copyright and licensing details.
```

**Comment by AntoineVDV** [comment]
@robodoo override=ci/style
> [odoo/.../static/demo/large_meeting_table_document.pdf](https://github.com/odoo/odoo/blob/281c7b6654e543f0de8ac2d5a6eea85bca1a7936/.../static/demo/large_meeting_table_document.pdf) is more than 100KB

Reason: Fabien a dit que je peux

File: addons/sale_pdf_quote_builder/models/ir_actions_report.py [hunk 4]
```diff
@@ -24,50 +24,105 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
...
-                try:
-                    form_fields = self._get_form_fields_mapping(order)
-                    stream = io.BytesIO(pdf.merge_pdf(pdf_data, form_fields))
-                    result[order.id].update({'stream': stream})
-                except (ValueError, PdfStreamError, PdfReadError, TypeError, zlib_error,
-                        NotImplementedError):
-                    order._message_log(body=_(
-                        "There was an error when trying to merge headers and footers to the "
-                        "original PDF.\n Please make sure the source file are valid."
-                    ))
```
Comment (Feyensv) [inline addons/sale_pdf_quote_builder/models/ir_actions_report.py:65]: No more feedback on errors ?

Comment (Demesmaeker) [reply addons/sale_pdf_quote_builder/models/ir_actions_report.py:65]: This was added when the feature was bugged + when it was still possible to put a non pdf file, which crashed at this point. But I only had to cover one method with this try except. Now the filling in the forms with so information, or sol information, or merging it is separated, adding this error every time is quite heavy, and I'm not sure of the added value of this.  There could still be some errors, obviously, but I'm not sure which ones I should track to cover everything related, and I'm a bit afraid to cover too much and silence something that I shouldn't.

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
Are you really considering sending a 2.4 Mb document as demo data (by the way, it is quite badly generated) ? Why don't you make a png image of this document that has anyway no added value ? This bloats the repo for nothing ...

**Comment by Feyensv** [comment]
@dalaOdoo can't you make a lighter version please ?
(or we could drop the file too :innocent:)

**Comment by mart-e** [comment]
The other two files in `addons/product/static/demo/` are in the 100-150k while containing +/- the same content. Not sure how this pdf was produced but it should also be possible to reduce it.

**Comment by Feyensv** [comment]
robodoo r-

**Comment by Demesmaeker** [comment]
Latest decision from above: not testing the feature improvement with a demo file, keep the previous one. So I took off the diff on this file.

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by AntoineVDV** [comment]
giphy

**Comment by Feyensv** [comment]
@robodoo retry

**Pull request closed without merging**